src/panoptes/pocs/camera/camera.py

```diff
@@ -636,6 +664,97 @@ def get_thumbnail(self, seconds, file_path, thumbnail_size, keep_file=False, *ar
             self.logger.warning(f'Problem getting thumbnail: {e!r}')
         return thumbnail
 
+    def _check_temperature_stability(self, required_stable_time=60*u.second,
+                                     sleep_delay=5*u.second, timeout=300*u.second,
+                                     blocking=False):
+        """
+        Wait until camera temperature is within tolerance for a sufficiently long period of time.
+
+        Args:
+            required_stable_time (astropy.units.Quantity): Minimum consecutive amount of
+                time to be considered stable. Default 60s.
+            sleep_delay (astropy.units.Quantity): Time to sleep between checks. Default 10s.
+            timeout (astropy.units.Quantity): Time before Timeout error is raised. Default 300s.
+            blocking (bool): Block until stable temperature or timeout? Useful for testing.
+        """
+        # Convert all times to seconds
+        sleep_delay = get_quantity_value(sleep_delay, u.second)
+        required_stable_time = get_quantity_value(required_stable_time, u.second)
+        timeout = get_quantity_value(timeout, u.second)
+
+        # Define an inner function to run in a thread
+        def check_temp(required_stable_time, sleep_delay, timeout):
+            if required_stable_time > timeout:
+                raise ValueError("required_stable_time must be less than timeout.")
+            if sleep_delay > timeout:
+                raise ValueError("sleep_delay must be less than timeout.")
+
+            # Wait until stable temperature persists or timeout
+            time_stable = 0
+            timer = CountdownTimer(duration=timeout)
+            while True:
+                if timer.expired():
+                    break
+                # We may need to restart the thread before it has finished
```

Comment:
Is this true?  When and why do we need to restart?

```suggestion
                # We may need to restart the thread before it has finished if another check has been requested.
```